blog/2022/storage-map.adoc

```diff
@@ -0,0 +1,89 @@
+with Keycloak 19, the preview of this storage is available in two 
+implementations: relational database and Infinispan.
+
+The new store supports only Postgres and CockroachDB databases. Legacy
+database is still available both with Wildfly and Quarkus distributions,
+and will be kept until the new store is production-ready and supports
+all first class databases.
+
+In addition to the persistent implementations, there is also an in-memory
+implementation of the new store used primarily for development purposes
+and incapable of zero-downtime upgradability. This implementation is based
+on Java native `ConcurrentHashMap`, and referred to by an abbreviation `chm`.
+
+We will be glad for you to try these feature out and let us know your
+feedback in https://github.com/keycloak/keycloak/discussions/new?category=keycloak-x-new-store&title=%5BUser%20feedback%5D%20&labels=team/storage-sig[GitHub Discussions].
+
+## Overview
+
+This section is an extract of the https://github.com/keycloak/keycloak-community/blob/main/design/keycloak.x/storage-architecture.md[architecture specification]. Please follow the link for more details.
+
+In Keycloak, the following storage areas have been identified: Realms,
+clients, users, groups, roles, client scopes, events, authentication
+sessions, user/client sessions, single-use tokens, and authorization
+services. 
+
+For each area, it is possible to specify a single storage. Yet each area
+is independent of others, so e.g. realms can be served by database and
+users by Infinispan.
+
+In Keycloak 19, the short CLI options support selecting
+one storage engine for all areas at once. Picking a different storage
+per area is only possible via generic options, see below. 
+In the future, it will be better supported to mix and match the storage
+implementations to tailor deployment to best match the target environment.
+
+Since each area is served by a single storage implementation, and the only
+available storages are database, external Infinispan, and `chm`, it is not
+possible to cache values, hence the performance is far from ideal.
+Ability to combine multiple stores in a single area will be delivered
+in Keycloak 20.
+
+== Setup Guides
+
+The following section contains examples of how to configure Keycloak 19
+with the new store.
+
+WARNING: The command line options starting with `--storage` might change
+         without prior warning, based on the feedback from the community
+         and project needs.
+
+=== Postgres / CockroachDB
+
+Requirements: Postgres 14 or CockroachDB 22.1 (or newer
+
+Run the following command:
+
+   bin/kc.sh start --storage=jpa --db-url=<jdbc-url> --db-username=<username> --db-password=<password>
+
+Reference to https://github.com/keycloak/keycloak/discussions/9596[Q&A live document].
+
+CAUTION: This feature has known issues in CockroachDB, see https://github.com/keycloak/keycloak/issues/11618[this tracker for list].
+
+=== External Infinispan
+
+Requirements: Infinispan 13
```

Comment:
This is not true. Currently, the HotRod store works only with Infinispan 12. We will work on Infinispan 13/14 soon, but it will require some changes in how we do the configuration.